pyaedt/modeler/cad/component_array.py

```diff
@@ -0,0 +1,843 @@
+
+        Returns
+        -------
+        str
+           Coordinate system name.
+        """
+        cs_dict = self._get_coordinate_system_id()
+        if self._cs_id not in cs_dict.values():
+            self._logger.warning("Coordinate system is not loaded, please save the project.")
+            return "Global"
+        else:
+            return [cs for cs in cs_dict if cs_dict[cs] == self._cs_id][0]
+
+    @coordinate_system.setter
+    def coordinate_system(self, name):
+        cs_dict = self._get_coordinate_system_id()
+        if name not in cs_dict.keys():
+            self._logger.warning("Coordinate system is not loaded, please save the project.")
+        else:
+            self._cs_id = cs_dict[name]
+            self._edit_array()
+
+    @property
+    def _array_props(self):
+        """Retrieve the properties of the component array.
+
+        Returns
+        -------
+        dict
+           An ordered dictionary of the properties of the component array.
+        """
+        return self.get_array_props()
+
+    @pyaedt_function_handler()
+    def delete(self):
+        """Delete the array.
+
+        References
+        ----------
+
+        >>> oModule.DeleteArray
+
+        """
+        self._app.omodelsetup.DeleteArray()
+        del self._app.component_array[self.name]
+        self._app.component_array_names = list(self._app.get_oo_name(self._app.odesign, "Model"))
+
+    @pyaedt_function_handler()
+    def export_array_info(self, array_path=None):
+        """Export array information to CSV file.
+
+        References
+        ----------
+
+        >>> oModule.ExportArray
+
+        """
+        if self._app.settings.aedt_version < "2024.1":  # pragma: no cover
+            self._logger.warning("This feature is not available in " + str(self._app.settings.aedt_version))
+            return False
+
+        if not array_path:  # pragma: no cover
+            array_path = os.path.join(self._app.toolkit_directory, "array_info.csv")
+        self._app.omodelsetup.ExportArray(self.name, array_path)
+        return True
+
+    @pyaedt_function_handler()
+    def get_array_props(self):
+        """Retrieve the properties of the component array.
+
+        Returns
+        -------
+        dict
+           An ordered dictionary of the properties of the component array.
+        """
+        # From 2024R1, array information can be loaded from a CSV
+        if self._array_info_path and os.path.exists(self._array_info_path):  # pragma: no cover
+            array_props = self.array_info_parser(self._array_info_path)
+        else:
+            self._app.save_project()
+            array_props = self._get_array_info_from_aedt()
+        return array_props
+
+    @pyaedt_function_handler()
+    def array_info_parser(self, array_path):  # pragma: no cover
+        """Parse array CSV file.
+
+        Returns
+        -------
+        dict
+           An ordered dictionary of the properties of the component array.
```

Comment:
```suggestion
           Ordered dictionary of the properties of the component array.
```
Is this the right description or an inadvertent copy-and-paste error from the previous entry?